File: 2020_11_09_reverse_array.js
// Reverse Array
// Given a numerical array, reverse the order of the
// values. The reversed array should have the same
// length, with existing elements moved to other
// indices so that the order of elements is reversed.
function reverseArr(arr) {
    let len = arr.length, temp = 0;
    for (let i = 0; i <= Math.floor((len-1)/2); i++) {
        temp = arr[i];
        arr[i] = arr[len-1-i];
        arr[len-1-i] = temp;
    }
    return arr;
}
console.log(reverseArr([1, 2, 3, 4, 5, 6]))
console.log(reverseArr([1, 2, 3, 4, 5]))

File: 2020_11_09_reverse_array.py
# Reverse Array
# Given a numerical array, reverse the order of the
# values. The reversed array should have the same
# length, with existing elements moved to other
# indices so that the order of elements is reversed.
import math
def reverseArr(li):
    l = len(li); temp = 0
    for i in range(0, math.floor((l-1)/2)+1):
        temp = li[i]
        li[i] = li[l-1-i]
        li[l-1-i] = temp
    return li;
print(reverseArr([1, 2, 3, 4, 5, 6]))
print(reverseArr([1, 2, 3, 4, 5]))